### ISSUE TITLE
dd support for unique phone numbers, find user by phone number in Identity

### ISSUE THREAD

**Issue by ayodejii** [dotnet/aspnetcore#55199]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

I am trying to create users with unique phone numbers.

I know it's already mentioned here in this issue [25368](https://github.com/dotnet/aspnetcore/issues/25368) and many others, but I think it makes sense to have it now.

### Describe the solution you'd like

As suggested in the issue I referenced above, I think there should be a provided flag when creating a user, to enforce this (or not).

`await userManager.CreateAsync(user, request.Password, requireUniquePhoneNumber = true) // default could be false`

### Additional context

In some countries, user identity is tied to their phone numbers, so some apps require users to register with them.

**Comment by ahmed-abdelrazek** [comment]
https://github.com/dotnet/aspnetcore/discussions/59444#discussioncomment-11916536

![image](https://github.com/user-attachments/assets/5c9f442d-db3c-48b6-89c1-d2cbf5d533fb)

**Comment by mguinness** [comment]
To clarify, previous poster is requesting an additional property be added to the [UserOptions](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.useroptions) class as part of this change.  Hard to discern from just a screenshot.

**Comment by ahmed-abdelrazek** [comment]
@mguinness yes it's my idea on how requiring unique phone number for user might work which is the same as how requiring unique email works
